### ISSUE TITLE
XCScheme File References Broken After Running Pod Install

### ISSUE THREAD

**Issue by jcolicchio** [CocoaPods/CocoaPods#10141]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

1. Add `Default Location`, `Routing App Coverage File`, and `StoreKit Configuration` to the `Run Options` of a new project's `Scheme` via `.xcodeproj`
2. Run `pod install`
3. View scheme in newly generated `.xcworkspace`

## What did you expect to happen?

Scheme preserves values originally selected in `.xcodeproj`

## What happened instead?

Values appear in red as broken references, except for `Default Location` which is set to `None`

### Stack

```
   CocoaPods : 1.9.3
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.7 (19H2)
       Xcode : 12.0 (12A7209)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 3751230cb29f78f77e1c450891179487873382c3

               trunk - CDN - https://cdn.cocoapods.org/
               udemy - git - git@github.com:udemy/cocoapods.git @ 44751c434ca7e662e70adfd3a33f0dd565622101

```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
claide-plugins         : 0.9.2
cocoapods-deintegrate  : 1.0.4
cocoapods-dependencies : 1.3.0
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.1.0
cocoapods-trunk        : 1.5.0
cocoapods-try          : 1.2.0
cocoapods_debug        : 0.1.0
slather                : 2.4.9
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'pod-scheme-location-thing' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for pod-scheme-location-thing

end
```

## Project that demonstrates the issue

https://github.com/jcolicchio/pod-scheme-location-thing

**Comment by jcolicchio** [comment]
This isn't a huge deal, but for projects which routinely generate `.xcscheme`s, it'd be nice not to have to manually fix the schemes every time

**Comment by dnkoutso** [comment]
I do not think CocoaPods changes the users schemes? Is this a scheme for the Pods.xcodeproj?

**Comment by jcolicchio** [comment]
I agree CocoaPods is not changing the scheme; I'm wondering if it should? Or if it should set things up such that the scheme continues to have valid values when being opened through the `.xcworkspace`.

A scheme can have a relative path to a `.gpx` file, but the path is relative to a different location depending on whether it is viewed by opening the `.xcodeproj` or the `.xcworkspace`

If you check out the attached example project and run `pod install`, you can open the `.xcworkspace`, edit the scheme and view `Run Options`, and you should see that the `Default Location` is `None`. If you then open the `.xcodeproj` and edit the same scheme, you'll see the `Default Location` is `Location`.

I'm wondering if there's a way to edit the scheme or the generated `.xcworkspace` so that the scheme is valid for both `.xcodeproj` and `.xcworkspace`

**Comment by dnkoutso** [comment]
Hmm if i understand correctly this is related to the .xcworkspace that cocoapods generates including the scheme there? You can always use a `post_install` hook to edit any scheme.

I should probably try the sample soon.

**Comment by jcolicchio** [comment]
So, I'm not sure there's anything we can really do to resolve this, short of modifying the user's xcscheme.

It comes down to the `LocationScenarioReference`'s `identifier` being a relative path that's relative to one of two different folders, depending on how the project was opened:

* `example.xcodeproj`: `example.xcodeproj/project.xcworkspace/.`
* `example.xcworkspace`: `example.xcworkspace/.`

Schemes need to use `../../` to get to the project root when opened via xcodeproj directly, but just `../` when the same scheme is used in a workspace. There's no value that can satisfy the scheme under both usages.

I've tried absolute paths, and I've tried $(PROJECT_DIR), but to no avail. At the end of the day, it seems like the scheme will have to have a valid value for one use-case, while having an invalid value for the other :/
